app/components/SavePublish/SavePublishWarnings.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from "react";
+import { Tag, App, FloatButton } from "antd";
+import { WarningFilled } from "@ant-design/icons";
+import { getRuleMap, generateSchemaFromRuleContent, updateRuleData } from "@/app/utils/api";
+import styles from "./SavePublish.module.css";
+
+interface SavePublishProps {
+  filePath: string;
+  ruleContent: { nodes: []; edges: [] };
```

Comment:
Could we update this to use the DecisionGraphType from "@gorules/jdm-editor", or its nested DecisionNode/DecisionEdge types?

Reply:
Good point. Updated!